syntax = "proto3";

package smart.camera;

import "google/protobuf/empty.proto";

message CameraRequest {
  string camera_id = 1;
}

message TogglePowerRequest {
  string camera_id = 1;
  bool turnOn = 2;
}

message MoveCameraRequest {
  string camera_id = 1;
  int32 angle = 2;
  int32 tilt = 3;
}

message CameraState {
  string camera_id = 1;
  bool is_on = 2;
  string type = 3;
  int32 angle = 4;
  int32 tilt = 5;
  string ip = 6;
}

message CameraInfo {
  string camera_id = 1;
  string type = 2;
  string ip = 3;
}

message CameraList {
  repeated CameraInfo cameras = 1;
}

message Ack {
  string message = 1;
}

service CameraService {
  rpc ListCameras (google.protobuf.Empty) returns (CameraList);
  rpc GetState (CameraRequest) returns (CameraState);
  rpc TogglePower (TogglePowerRequest) returns (Ack);
  rpc MoveCamera (MoveCameraRequest) returns (Ack);
}
